packages/core/src/internal/utils/future-id-builders.ts

```diff
@@ -4,6 +4,12 @@
  */
 const MODULE_SEPERATOR = "#";
 
+/**
+ * The separator in ids that depend on futures that belong to a submodule.
+ * This separator is used to split the submodule and the rest of the dependency's id.
+ */
+const SUBMODULE_SEPARATOR = "-";
```

Comment:
I accept better alternatives. I tried "##", but I think we'd regret it in the future.

Reply:
What do you think of `~`?

We allow the user to provide `-` for deployment ids, so I worry we would create ambiguity in parsing.

Alternatively we could ban `-` in deployment ids and all other ignition ids to ensure `-` only has the meaning _SUBMODULE_SEPARATOR_.

Reply:
That's a good point.

I hesitated about `~` initially, as it has some special meaning in UNIX systems, like `~`, `~pato`. But in our case it will always be in the middle of two words, so it shouldn't be a problem.